test/Regex/characterclass_with_range.js

```diff
@@ -0,0 +1,163 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+let re = /^[\s-a-z]$/;
+let reIgnoreCase = /^[\s-a-z]$/i;
+let reUnicode = /^[\s-z]$/u;
+let reNoCharClass = /^[a-c-z]$/;

```

Comment:
>/^[a-c-z]$/; [](start = 20, length = 12)

Remember that /^[blah]$/ produces a different MatchSet bytecode than /[blah]/ -- you might want to double check that the change to the parser logic creates the correct set in various bytecodes (since we're creating the set itself in the part of the parser we're looking at here, I don't see why it would be any different, but it's a good idea to check).